### PR TITLE
Some quality of life improvements

### PR DESCRIPTION
**Pull request by dorukkilitcioglu**
This PR does some housekeeping on minor bugfixes and enhancements. Specifically:

- Makes sure the SVD matrix is part of the model parameters and is fine-tuneable
- Updates how models should be persisted to make sure models that are saved on a GPU can be loaded on a CPU
- Adds new embeddings
- Adds pooling inside the word embedding, which decreases the memory footprint of the model compared to constructing word embeddings first and pooling them afterwards

I expect most changes to be straightforward apart from the last one. The specific scenario I've encountered is with trying to generate embeddings like `BERT-mean-SVD`, where there is a transformation after pooling:
```
tw = TextWiser(Embedding.Word(word_option=WordOptions.bert),
               [Transformation.Pool(pool_option=PoolOptions.mean), Transformation.SVD()])
```

When specified like this, TextWiser first transforms all documents into individual word embeddings, pools all words belonging to the same sentence, and does an SVD on top of the pooled documents. The problem occurs at the first step, where the individual word embeddings can be too large to fit in the GPU memory all at once.

With this PR, you can specify the following model, which significantly reduces that problem:
```
tw = TextWiser(Embedding.Word(word_option=WordOptions.bert, pool_option=PoolOptions.mean),
               Transformation.SVD())
```
This way, each document is pooled right after it's word embeddings are constructed, meaning you reduce the memory footprint of a single document's word embedding. The end result is the same as the above, but you can give more documents before you run out of GPU memory.

Let me know if the above description is clear and if I should add more details to the documentation to make that clear. Thanks!

### CODE REVIEW

File: textwiser/__init__.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 """
 
 __author__ = "FMR LLC"
-__email__ = "textwiser@fmr.com"
+__email__ = "opensource@fidelity.com"
```
Comment (erstrong) [inline textwiser/__init__.py:23]: @skadio should we also be using this email for mabwiser?

File: notebooks/persistence_example.ipynb [hunk 2]
```diff
@@ -23,9 +23,11 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Pickle\n",
+    "## `torch.save` and `torch.load`\n",
+    "\n",
+    "The most general-purpose way of saving TextWiser objects is to use [torch.save](https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-entire-model). This has the benefit of letting all Torch tensors be handled by PyTorch, while pickling everything else.\n",
```
Comment (erstrong) [inline notebooks/persistence_example.ipynb:28]: So the change for GPU vs CPU persistence is just to use a different external-to-textwiser protocol?

**Review by erstrong — APPROVED** [review]
Looks good!

**Review by erstrong — APPROVED** [review]
Looks good!

File: notebooks/persistence_example.ipynb (hunk 2 above)
Comment (dorukkilitcioglu) [reply notebooks/persistence_example.ipynb:28]: Yup! There wasn't an explicit `save` function for the TextWiser object anyway, and these new functions reuse the existing functionality that enables `load_state_dict` and `save_state_dict` to work while being simpler to use.

File: textwiser/__init__.py (hunk 1 above)
Comment (skadio) [reply textwiser/__init__.py:23]: what's this email?

Comment (skadio) [reply textwiser/__init__.py:23]: why not get rid of all email communication and depend on GitHub issues solely?

File: textwiser/embeddings/word.py [hunk 3]
```diff
@@ -128,13 +146,14 @@ def _get_and_init_word_embeddings(word_option: WordOptions, pretrained: str, **p
 
 class _WordEmbeddings(BaseFeaturizer):
     def __init__(self, word_option: WordOptions, pretrained=Constants.default_model, sparse=True, tokenizer=None,
-                 layers=-1, **kwargs):
+                 layers=-1, pool_option: Optional[PoolOptions] = None, **kwargs):
```
Comment (skadio) [inline textwiser/embeddings/word.py]: I understand the improvement in memory footprint. But this also changes the API so I am not sure how to feel about it. 

So the new pool_option is only available for WordEmbeddings? 

Previously we separated Embeddings from Transformations --which is invariant/true for ALL methods. 

Here you are introducing a case where we are mixing Embeddings with inline transformation, which goes against the invariant above.

What happens if some use both of them at the same time? For example, what's the semantics of a call like: 

emb = Textwiser(Emebdding.Word(bert, pool = mean) + Transformation.pool(mean) ) 

Have you considered an alternative, such as a special pooling option for Word embeddigns? Sth like 

emb = Textwiser(Emebdding.Word(bert) + Transformation.pool(mean_per_word) ) 

(I made up the mean_per_word naming, but you got the idea)

File: textwiser/embeddings/word.py [hunk 4]
```diff
@@ -128,13 +146,14 @@ def _get_and_init_word_embeddings(word_option: WordOptions, pretrained: str, **p
...
         super(_WordEmbeddings, self).__init__()
         self.word_option = word_option
         self.pretrained = pretrained
         self.sparse = sparse
         self.tokenizer = tokenizer if tokenizer else split_tokenizer
         self.layers = [layers] if isinstance(layers, int) else layers
+        self.pool_option = pool_option
```
Comment (skadio) [inline textwiser/embeddings/word.py]: not sure about this api --see comments above

File: textwiser/utils.py [hunk 5]
```diff
@@ -4,7 +4,6 @@
 
 """
 :Author: FMR LLC
-:Email: textwiser@fmr.com
```
Comment (skadio) [inline textwiser/utils.py:7]: nice!

**Comment by skadio** [comment]
Looks good to me. One comment about API

**Review by skadio — APPROVED** [review]

**Comment by skadio** [comment]
Not immediately related to this PR, but if you are doing housekeeping.. 

I was going thru the notebooks, I see some comments like "here is the old way of doing it".. 
I don't think it would be obvious to anyone what that refers to. may be remove those?

File: textwiser/embeddings/word.py (hunk 3 above)
Comment (dorukkilitcioglu) [reply textwiser/embeddings/word.py]: I'm updating this based on our discussion:

- The naming will be switched to `inline_pool_option` to differentiate it from pool transformation
- The documentation will be super clear about how you should only use the inline pooling and the pool transformation, never both
- The TextWiser object will throw a specific error if you ignore the documentation and do it anyway

If there ends up being other cases like this future steps (transformations) can influence how we calculate the previous steps (embeddings), we can start thinking about a more general purpose solution.

Comment (skadio) [reply textwiser/embeddings/word.py]: Sounds good. I like the idea of starting small, testing out the approach, and eventually, we might be able to generalize from here. It seems more "usage" is necessary to understand a pattern here.

File: textwiser/embeddings/word.py (hunk 4 above)
Comment (dorukkilitcioglu) [reply textwiser/embeddings/word.py]: Replied above!

**Comment by dorukkilitcioglu** [comment]
Agreed on that "old way" comment as well. If you're directly using the implementation objects something has gone horribly wrong in the documentation...

I'll remove them.

**Pull request merged**